[24.7 PiT]  gradle project build failed with 24.7-SNAPSHOT in production mode

### Describe the bug

when build the gradle project with current 24.7-SNAPSHOT(b004eb0b7f8afc8cff095638c393dcc122b8d5e8), i got the following error in `skeleton-starter-hilla-lit-gradle` and `skeleton-starter-hilla-react-gradle`

```
> Task :vaadinBuildFrontend FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':vaadinBuildFrontend'.
> Vite process exited with non-zero exit code.
  Stderr: 'vite v6.0.6 building for production...
  transforming...
  ✓ 330 modules transformed.
  rendering chunks...
  computing gzip size...
  ../../../build/resources/main/META-INF/VAADIN/webapp/index.html                             0.70 kB │ gzip:   0.42 kB
  ../../../build/resources/main/META-INF/VAADIN/webapp/VAADIN/build/indexhtml-DHY83LLX.css    0.74 kB │ gzip:   0.33 kB
  ../../../build/resources/main/META-INF/VAADIN/webapp/VAADIN/build/about-view-7WFWUWuc.js    0.71 kB │ gzip:   0.51 kB
  ../../../build/resources/main/META-INF/VAADIN/webapp/VAADIN/build/indexhtml-B6m_yG3J.js   614.04 kB │ gzip: 157.97 kB

  (!) Some chunks are larger than 500 kB after minification. Consider:
  - Using dynamic import() to code-split the application
  - Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
  - Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
  src/main/frontend/views/helloworld/hello-world-view.ts(32,23): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string | TemplateResult'.
    Type 'undefined' is not assignable to type 'string | TemplateResult'.
  '

```

the error referred code is the `Notification.show(serverResponse)`
```
<Button
          onClick={async () => {
            const serverResponse = await HelloReactEndpoint.sayHello(name);
            Notification.show(serverResponse); 
          }}
        >
```

### Expected-behavior

no error in the build log

### Reproduction

- git clone git@github.com:vaadin/skeleton-starter-hilla-lit-gradle.git or git@github.com:vaadin/skeleton-starter-hilla-react-gradle.git 
- update the `vaadinVersion` in the gradle.properties to 24.7-SNAPSHOT
- ./gradlew clean build -Pvaadin.productionMode

### System Info

vaadin 24.7

--- a/packages/java/engine-core/src/main/java/com/vaadin/hilla/engine/AotBrowserCallableFinder.java
+++ b/packages/java/engine-core/src/main/java/com/vaadin/hilla/engine/AotBrowserCallableFinder.java
@@ -152,20 +152,19 @@ class AotBrowserCallableFinder {
         var annotationNames = engineConfiguration.getParser()
                 .getEndpointAnnotations().stream().map(Class::getName).toList();
 
-        try (var classLoader = new URLClassLoader(urls,
-                AotBrowserCallableFinder.class.getClassLoader())) {
-            return candidates.stream().map(name -> {
-                try {
-                    return Class.forName(name, false, classLoader);
-                } catch (Throwable t) {
-                    return null;
-                }
-            }).filter(Objects::nonNull)
-                    .filter(cls -> Arrays.stream(cls.getAnnotations())
-                            .map(Annotation::annotationType).map(Class::getName)
-                            .anyMatch(annotationNames::contains))
-                    .collect(Collectors.toList());
-        }
+        var classLoader = new URLClassLoader(urls,
+                AotBrowserCallableFinder.class.getClassLoader());
+        return candidates.stream().map(name -> {
+            try {
+                return Class.forName(name, false, classLoader);
+            } catch (Throwable t) {
+                return null;
+            }
+        }).filter(Objects::nonNull)
+                .filter(cls -> Arrays.stream(cls.getAnnotations())
+                        .map(Annotation::annotationType).map(Class::getName)
+                        .anyMatch(annotationNames::contains))
+                .collect(Collectors.toList());
     }
 
     private static String quotePath(Path path) {

